server/pulp/plugins/distributor.py

```diff
@@ -187,6 +191,20 @@ def create_consumer_payload(self, repo, config, binding_config):
         """
         return {}
 
+    @staticmethod
+    def ensure_all_units_downloaded(repo_id):
+        """
+        Checks the database to make sure all units in the repo have been downloaded. If not, raises
+        an exception.
+
+        :param repo_id: repository id
+        :type  repo_id: basestring
+
+        :raises exceptions.PulpCodedException:  if any unit in the repo has un-downloaded files
```

Comment:
Both I guess? I think a unit is "undownloaded" if it has one or more files that are undownloaded. Do you think there's a way to make the comment here more clear?